#3464 display data on bar chart

mark-prins opened the pull request:
Fixes #3464 

## Change summary

Have added three bars to display the hot-mid-cold colour bars.
To do so, I've added a selector to grab the upper and lower breach values - it was unclear what, from the mockup, should be used out of the consecutive and cumulative options. ideas?

This version replaces the current line chart - should we have both options available?

This is how it looks with the generated data:

<img width="1219" alt="Screen Shot 2021-02-05 at 1 13 31 PM" src="https://user-images.githubusercontent.com/9192912/106971875-a4b48400-67b4-11eb-807d-b56a86c9d582.png">


## Testing

Steps to reproduce or otherwise test the changes of this PR:

- [ ] View fridge detail
- [ ] adjust the breach thresholds and check the chart again

### Related areas to think about

If there are any general areas of the codebase your changes might have side affects on, mention them here

joshxg reviewed (approved):
Wow that is looking great! When I played around I was trying to make the red/blue appear as the whisker - this seems much nicer!

> should be used out of the consecutive and cumulative options. ideas?

Consecutive, I reckon

> This version replaces the current line chart - should we have both options available?

I.. guess..? I dunno either..

- Wondering if the hazard icons aren't needed? Think it looks pretty great without? Could make the red/blue parts pressable (I think you can pass in a custom component?) instead?  (new issue?)
- Also just occurred to me about what happens when the breach thresholds change. At the moment we're calculating the breach thresholds from the current configs, but what if we had breaches a month ago at 8 degrees, and then the threshold was changed to 12 - we keep the thresholds on the `TemperatureBreach` record for this reason so we historically know what the thresholds were, but I'm not sure what we should do about this on the chart - maybe this is a good reason to keep the line chart around - easier to handle this with breach icons on a line chart? 🤣 

Also couple other things we should make issues for:
- Fix the date/time ticks along the x axis to align with the bars. Should also change them for the line chart to make them smaller and more frequent- adding some time in when theres more than one tick for the same day
- Grid lines like on the marvel? Think they look pretty nice

mark-prins commented:
I've added the grid lines and changed to use consecutive thresholds and removed breach hazard notification icons from the bar chart.

then went offroad.. there's now an option to select the chart type (top right)


<img width="1214" alt="Screen Shot 2021-02-05 at 5 21 25 PM" src="https://user-images.githubusercontent.com/9192912/106989791-c8d58c80-67d7-11eb-9e13-098a5de3a6db.png">

<img width="1220" alt="Screen Shot 2021-02-05 at 5 21 32 PM" src="https://user-images.githubusercontent.com/9192912/106989794-ca9f5000-67d7-11eb-9a23-d2f31edb4f5c.png">

sorry?

joshxg commented:
So coool!!! ❤️
